modules/tribes/client/components/CirclePage.component.js

```diff
@@ -0,0 +1,190 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { useTranslation } from 'react-i18next';
+
+import * as api from '../api/tribes.api';
+import JoinButton from './JoinButton';
+import { getRouteParams } from '@/modules/core/client/services/angular-compat';
+
+export default function CirclePage({ user, goBack }) {
+  const [circle, setCircle] = useState(null);
+  const { t } = useTranslation('tribes');
+
+  const fetchData = async () => {
+    const circle = await api.get(getRouteParams().circle);
+    setCircle(circle);
+  };
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  // Circle loading
+  if (!circle) {
+    return (
+      <div className="container container-spacer text-muted text-center">
+        {/* <!-- tr-spinner will have `aria-busy` etc --> */}
+        {/* <tr-spinner size="sm"></tr-spinner> */}
+        <br />
+        <br />
+        <small aria-hidden="true">{t('Wait a moment...')}</small>
+      </div>
+    );
+  }
+
+  // Circle not found
+  if (!circle._id) {
+    return (
+      <section className="container container-spacer">
+        <div className="row">
+          <div className="col-xs-12">
+            <h2>{t('This circle is not here...')}</h2>
+            <p className="lead">
+              {t(
+                `The circle you're seeking isn't here. Check if your address is correct.`,
+              )}
+            </p>
+            <a onClick={goBack} className="btn btn-primary">
+              {t('See other circles')}
+            </a>
+          </div>
+        </div>
+      </section>
+    );
+  }
+
+  // Circle found
+  return (
+    <section
+      className={`board tribe-image tribe-header ${!user ? 'is-guest' : ''}`}
```

Comment:
We've used [`classnames`](https://www.npmjs.com/package/classnames) helper to simplify these conditional checks, [example](https://github.com/Trustroots/trustroots/blob/5711b1da0b346a90f5f691e5d9c831ff40c36e6d/modules/core/client/components/Icon.js#L9-L16).